crypto/s2n_hash.c

```diff
@@ -22,6 +22,45 @@
 
 #include "utils/s2n_safety.h"
 
+#if S2N_OPENSSL_VERSION_AT_LEAST(1,1,1) || defined(OPENSSL_IS_BORINGSSL) || defined(OPENSSL_IS_AWSLC)
```

Comment:
Can we do a feature probe for this? Just try to compile a file that refers to the symbol and if it all links then we know we have it and we set the `S2N_EVP_SUPPORTS_SHA1_MD5_HASH` in the build.

Reply:
I love this idea, but none of our current feature probes involve libcrypto.

It took a bit of finagling to get it working, and unless you think it's a blocker I'd like to submit it in a separate PR. I want a separate review very clearly focused on those cmake/make changes.